### PR TITLE
Run custom build scripts

### PR DESCRIPTION
**Pull request by Skrupellos**
You can add a texlive.run-file, which will be run after installation of all dependencies. Here you could compile your tex-document and do something with the result.

### CODE REVIEW

**Comment by syphar** [comment]
@Skrupellos thanks for this :) 

What's your usecase for this? what are you using it for?

**Comment by Skrupellos** [comment]
We build our tex files (which is a little bit complicated xD) with heroku and push the result to our publishing server: https://github.com/LMUGAF/ersti-einstein/blob/master/texlive.run
Or in short: CI

**Comment by codingjoe** [comment]
I think it makes more sense to do it similar to the official build packs. I remember python using `bin/post_compile`.

@Skrupellos I don't get yet what you mean with
>build our tex files
How do I have to understand that?

**Comment by syphar** [comment]
@codingjoe it would definitely have to be a different naming, for those using multiple buildpacks. 
but I agree we can improve on naming the executable :)

**Comment by Skrupellos** [comment]
@codingjoe 
>>  build our tex files
> How do I have to understand that?
1. User pushes to GitHub
2. Heroku compiles tex to pdf (see https://github.com/LMUGAF/ersti-einstein/blob/master/texlive.run)
3. Heroku pushes the resulting pdf to a server (also in texlive.run, but you can also publish it on GitHub or where ever)
... this is on of the most simple and basic use cases: Push your changes and download the resulting PDF $here.

@syphar: Naming suggestions?

File: bin/compile [hunk 1]
```diff
@@ -107,3 +108,11 @@ fi
 
 # Set up the environment for runtimes now that compilation has finished
 echo 'export PATH=$HOME/.texlive/bin/x86_64-linux:$PATH' >> $PROFILE_D
+
+# run custom job
+if [ -f $BUILD_DIR/texlive.run ]; then
+    build-step "run custom (build-)script"
+    chmod +x $BUILD_DIR/texlive.run
+    . $BUILD_DIR/texlive.run
```
Comment (codingjoe) [inline bin/compile]: wouldn't it make more sense to source the script?

File: bin/compile [hunk 2]
```diff
@@ -107,3 +108,11 @@ fi
 
 # Set up the environment for runtimes now that compilation has finished
 echo 'export PATH=$HOME/.texlive/bin/x86_64-linux:$PATH' >> $PROFILE_D
+
+# run custom job
+if [ -f $BUILD_DIR/texlive.run ]; then
```
Comment (codingjoe) [inline bin/compile]: I don't like the `.run`-extension. Heroku doesn't do that in their own build packs, so i'd like to keep it consistent. No extension should be fine.
Maybe you this should go into the `bin/` folder as it is done for the python buildpack?

**Comment by codingjoe** [comment]
Since there hasn't been any update for month, I will close this PR for now. Feel free to reopen it at any time. Thanks :)

**Pull request closed without merging**